runtime/kubernetes/build.go

```diff
@@ -39,22 +40,88 @@ func (c *client) InspectBuild(ctx context.Context, b *pipeline.Build) ([]byte, e
 func (c *client) SetupBuild(ctx context.Context, b *pipeline.Build) error {
 	c.Logger.Tracef("setting up for build %s", b.ID)
 
+	if c.PipelinePodTemplate == nil {
+		if len(c.config.PipelinePodsTemplateName) > 0 {
+			// nolint: contextcheck // ignore non-inherited new context
+			podsTemplateResponse, err := c.VelaKubernetes.VelaV1alpha1().PipelinePodsTemplates(c.config.Namespace).Get(
+				context.Background(), c.config.PipelinePodsTemplateName, metav1.GetOptions{},
+			)
+			if err != nil {
+				return err
+			}
+
+			// save the PipelinePodTemplate to use later in SetupContainer and other Setup methods
+			c.PipelinePodTemplate = &podsTemplateResponse.Spec.Template
+		} else {
+			c.PipelinePodTemplate = &v1alpha1.PipelinePodTemplate{}
+		}
+	}
+
+	// These labels will be used to call k8s watch APIs.
+	labels := map[string]string{"pipeline": b.ID}
+
+	if c.PipelinePodTemplate.Meta.Labels != nil {
+		// merge the template labels into the worker-defined labels.
+		for k, v := range c.PipelinePodTemplate.Meta.Labels {
+			// do not allow overwriting any of the worker-defined labels.
+			if _, ok := labels[k]; ok {
+				continue
+			}
+
+			labels[k] = v
+		}
+	}
+
 	// create the object metadata for the pod
 	//
 	// https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1?tab=doc#ObjectMeta
 	c.Pod.ObjectMeta = metav1.ObjectMeta{
-		Name:   b.ID,
-		Labels: map[string]string{"pipeline": b.ID},
+		Name:        b.ID,
+		Labels:      labels,
+		Annotations: c.PipelinePodTemplate.Meta.Annotations,
+	}
+
+	// TODO: Vela admin defined worker-specific: AutomountServiceAccountToken
```

Comment:
I did not include `AutomountServiceAccountToken` in this PR as it requires more thought. The use of this might need to change when I add a feature that I haven't filed an issue for yet (The k8s runtime will need to query the docker registry for images, which might mean it needs to do something different with the ServiceAccount and any ImagePullSecrets.)